sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -1892,6 +1932,11 @@ object AnsiCast {
 
     case (StringType, _: CalendarIntervalType) => true
 
+    case (LongType, DayTimeIntervalType) => true
+    case (DayTimeIntervalType, LongType) => true
+    case (IntegerType, YearMonthIntervalType) => true
+    case (YearMonthIntervalType, IntegerType) => true
```

Comment:
We should allow casting to any numeric type according to the SQL standard. Am I right? @gengliangwang @srielau @cloud-fan